### PR TITLE
update api types, remove meta peer by id

### PR DESCRIPTION
**Pull request by ripienaar**
Signed-off-by: R.I.Pienaar <rip@devco.net>

/cc @nats-io/core

### CODE REVIEW

File: server/jetstream_cluster_test.go [hunk 1]
```diff
@@ -2941,7 +2941,7 @@ func TestJetStreamClusterPeerRemovalAPI(t *testing.T) {
 		t.Fatalf("Unexpected error: %v", err)
 	}
 
-	req = &JSApiLeaderServerRemoveRequest{Server: "S-2"}
+	req = &JSApiMetaServerRemoveRequest{Server: c.servers[1].ID()}
```
Comment (derekcollison) [inline server/jetstream_cluster_test.go:2944]: Maybe do `c.serverByName("S-2").ID()`

**Review by derekcollison — APPROVED** [review]
LGTM - Minor suggestion on test.

File: server/jetstream_cluster_test.go (hunk 1 above)
Comment (ripienaar) [reply server/jetstream_cluster_test.go]: ah, missed that method, nice.

**Comment by ripienaar** [comment]
Had one green build, will merge.

**Pull request merged**